pudb/var_view.py

```diff
@@ -389,142 +611,101 @@ def walk_value(self, parent, label, value, id_path=None, attr_prefix=None):
                                 + " (!! %s error !!)" % iinfo.display_type
                 ui_log.exception("stringifier failed")
 
-            if iinfo.show_detail:
-                if iinfo.access_level == "public":
-                    marker = "pub"
-                elif iinfo.access_level == "private":
-                    marker = "pri"
-                else:
-                    marker = "all"
-                if iinfo.show_methods:
-                    marker += "+()"
-                displayed_value += " [%s]" % marker
-
-            new_parent_item = self.add_item(parent, label, displayed_value,
-                id_path, attr_prefix)
-
-            if not iinfo.show_detail:
-                return
-
-            # set ---------------------------------------------------------
-            if isinstance(value, (set, frozenset)):
-                for i, entry in enumerate(value):
-                    if i % 10 == 0 and i:
-                        cont_id_path = "%s.cont-%d" % (id_path, i)
-                        if not self.frame_var_info.get_inspect_info(
-                                cont_id_path, read_only=True).show_detail:
-                            self.add_item(new_parent_item, "...", None,
-                                cont_id_path)
-                            break
-
-                    self.walk_value(new_parent_item, None, entry,
-                        "%s[%d]" % (id_path, i))
-                if not value:
-                    self.add_item(new_parent_item, "<empty>", None)
-                return
-
-            # containers --------------------------------------------------
-            key_it = None
+        if iinfo.show_detail:
+            if iinfo.access_level == "public":
+                marker = "pub"
+            elif iinfo.access_level == "private":
+                marker = "pri"
+            else:
+                marker = "all"
+            if iinfo.show_methods:
+                marker += "+()"
+            displayed_value += " [%s]" % marker
 
-            try:
-                key_it = value.keys()
-            except AttributeError:
-                # keys or iterkeys doesn't exist, not worth mentioning!
-                pass
-            except Exception:
-                ui_log.exception("Failed to obtain key iterator")
+        new_parent_item = self.add_item(parent, label, displayed_value,
+            id_path, attr_prefix)
+
+        if not iinfo.show_detail:
+            return
+
+        # containers --------------------------------------------------
```

Comment:
I've come to like vim's fold-y comments, i.e.
```suggestion
        # {{{ show container contents
```
(with corresponding closing comment in the appropriate spot of course)

Reply:
Ooh yes those are snazzy! In this case though, I think what really needs to happen is for these long subsections to be extracted into their own methods.

Reply:
:+1: